tests/FakeItEasy.Specs/AssignsOutAndRefParametersSpecs.cs

```diff
@@ -2,122 +2,169 @@ namespace FakeItEasy.Specs
 {
     using System;
     using System.Diagnostics.CodeAnalysis;
+    using FakeItEasy.Configuration;
     using FakeItEasy.Tests.TestHelpers;
     using FluentAssertions;
     using Xbehave;
 
-    public interface IHaveAnOut
+    public delegate void VoidDelegateWithOutAndRefParameters(int x, ref int y, out int z);
+
+    public delegate int NonVoidDelegateWithOutAndRefParameters(int x, ref int y, out int z);
+
+    public interface IHaveARef
     {
         [SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "0#", Justification = "Required for testing.")]
-        void MightReturnAKnownValue(out string andThisIsWhoReallyDidIt);
+        void MightReturnAKnownValue(ref string andThisIsWhoReallyDidIt);
     }
 
     public static class AssignsOutAndRefParametersSpecs
     {
+        const string Condition = "someone_else";
+        const string KnownOutput = "you";
+
         [Scenario]
         public static void AssignOutAndRefParametersLazilyUsingFunc(
-            IHaveAnOut subject,
+            IHaveARef subject,
             string outValue)
         {
-            var condition = "someone_else";
-            var knownOutput = "you";
+            "Given a fake IHaveARef"
+                .x(() => subject = A.Fake<IHaveARef>());
 
-            "establish"
-                .x(() => subject = A.Fake<IHaveAnOut>());
-
-            "when configuring a fake to assign out and ref parameters lazily using func"
-                .x(() => A.CallTo(() => subject.MightReturnAKnownValue(out outValue))
+            "When the fake is configured to assign out and ref parameters lazily using func"
+                .x(() => A.CallTo(() => subject.MightReturnAKnownValue(ref outValue))
                     .WithAnyArguments()
                     .AssignsOutAndRefParametersLazily((string value) => new object[]
                     {
-                        value == condition ? knownOutput : "me"
+                        value == Condition ? KnownOutput : "me"
                     }));
 
-            "it should assign the conditional value to the out field"
+            "And the configured method is called with matching arguments"
                 .x(() =>
-                    {
-                        string value = condition;
-                        subject.MightReturnAKnownValue(out value);
-                        value.Should().BeEquivalentTo(knownOutput);
-                    });
+                {
+                    outValue = Condition;
+                    subject.MightReturnAKnownValue(ref outValue);
+                });
+
+            "Then it should assign the conditional value to the ref field"

```

Comment:
Based on recent comments in my property setter issue, I think @adamralph would push for something more like "Then is assigns the conditional value to the ref field". But he's not here.